src/ng/filter/filters.js

```diff
@@ -221,18 +221,32 @@ function roundNumber(parsedNumber, fractionSize, minFrac, maxFrac) {
     var digit = digits[roundAt];
 
     if (roundAt > 0) {
-      digits.splice(roundAt);
+      // Drop fractional digits beyond `roundAt`
+      digits.splice(Math.max(parsedNumber.i, roundAt));
+
+      // Set non-fractional digits beyond `roundAt` to 0
+      for (var j=roundAt; j < digits.length; j++) {
+        digits[j] = 0;
+      }
     } else {
       // We rounded to zero so reset the parsedNumber
```

Comment:
Inserted the following line here, solves problem (B) described in https://github.com/angular/angular.js/pull/13913/files#r51409840:

```js
fractionLen = Math.max(0, fractionLen);
```